frontend/packages/kubevirt-plugin/src/components/vm-templates/vm-template-create-yaml.tsx

```diff
@@ -22,13 +23,38 @@ const CreateVMTemplateYAMLConnected = connectToPlural(
       }
       return <ErrorPage404 />;
     }
+    const [selectedTemplate, setTemplate] = React.useState(null);
+    const [templateLabel, setTemplateLabel] = React.useState(null);
 
     const template = VMTemplateYAMLTemplates.getIn(['vm-template']);
     const obj = safeLoad(template);
     obj.kind = kindObj.kind;
     obj.metadata = obj.metadata || {};
     obj.metadata.namespace = match.params.ns || 'default';
 
+    if (templateLabel) {
+      obj.objects[0].metadata.labels['vm.kubevirt.io/template'] = templateLabel;
+    }
+
+    if (!selectedTemplate) {
+      k8sGet(TemplateModel, '', 'openshift')
+        .then((templates) => {
+          setTemplate(
+            templates.items.filter(
+              (item) =>
+                item.metadata.labels['flavor.template.kubevirt.io/medium'] &&
+                item.metadata.labels['workload.template.kubevirt.io/server'] &&
+                item.metadata.name.includes('fedora'),
+            )[0],
+          );
+        })
+        .catch((err) => console.log('err:', err));
+    }
+
+    if (selectedTemplate && !templateLabel) {
```

Comment:
why different if for set and then using the templateLabel? I don't think there needs to be a state for the labels since the template includes it

Reply:
also please use selectors

Reply:
`selectVM` and `getLabels`

Reply:
> also please use selectors

I don't see how it will improve something. it will make the code less readable IMO.

Reply:
it will make it more specific and less error prone - null checks and vm selection process